server/app/assets/javascripts/main.ts

```diff
@@ -363,6 +363,18 @@ function attachRedirectToPageListeners() {
   })
 }
 
+/**
+ * Adds listeners to buttons with 'form' attributes. These buttons trigger form
+ * submission without JS. But we still need to stop propagation of events
+ * because it's possible that some other button up-stream in ancestor chain
+ * contains click listener as we have nested clickable elements.
+ */
+function attachListenersOnFormButtons() {
```

Comment:
Rename this for what the attached listener does, the naming is too vague as anything could be happening in the listener, without clarity in the name.

EG: attachStopPropogationListenerOnFormButtons.